staking/types/validator.go

```diff
@@ -324,16 +325,18 @@ func (w *ValidatorWrapper) SanityCheck(
 	switch len(w.Delegations) {
 	case 0:
 		return errors.Wrapf(
-			errInvalidSelfDelegation, "no self delegation given at all",
+			ErrInvalidSelfDelegation, "no self delegation given at all",
```

Comment:
Just return this error value unwrapped

Reply:
wrapping is necessary to provide more information.